workers/unity/Packages/com.improbable.gdk.health/Systems/HealthRegenSystem.cs

```diff
@@ -131,75 +126,63 @@ private void ApplyHealthRegen()
-                            healthRegenComponentData[i] = regenComponent;
-                        }
+                        return;
                     }
 
-                    healthRegenData[i] = regenData;
-
-                    return;
-                }
-
-                // If not damaged recently, and not already fully healed, regen.
-                if (healthComponent.Health < healthComponent.MaxHealth)
-                {
-                    regenData.NextRegenTimer -= Time.deltaTime;
-                    if (regenData.NextRegenTimer <= 0)
+                    // If damaged recently, tick down the timer.
+                    if (regenComponent.DamagedRecently)
                     {
-                        regenData.NextRegenTimer += regenComponent.RegenInterval;
+                        regenData.DamagedRecentlyTimer -= Time.deltaTime;
 
-                        // Send command to regen entity.
-                        var modifyHealthRequest = new HealthComponent.ModifyHealth.Request(
-                            spatialIdData[i].EntityId,
-                            new HealthModifier()
+                        if (regenData.DamagedRecentlyTimer <= 0)
+                        {
+                            regenData.DamagedRecentlyTimer = 0;
+                            regenComponent.DamagedRecently = false;
+                            regenComponent.RegenCooldownTimer = 0;
+                        }
+                        else
+                        {
+                            // Send a spatial update once every CooldownSyncInterval.
+                            regenData.NextSpatialSyncTimer -= Time.deltaTime;
+                            if (regenData.NextSpatialSyncTimer <= 0)
                             {
-                                Amount = regenComponent.RegenAmount
+                                regenData.NextSpatialSyncTimer += regenComponent.CooldownSyncInterval;
+                                regenComponent.RegenCooldownTimer = regenData.DamagedRecentlyTimer;
                             }
-                        );
-                        commandSystem.SendCommand(modifyHealthRequest);
+                        }
+
+                        return;
                     }
 
-                    healthRegenData[i] = regenData;
-                }
-            }
+                    // If not damaged recently, and not already fully healed, regen.
+                    if (healthComponent.Health < healthComponent.MaxHealth)
+                    {
+                        regenData.NextRegenTimer -= Time.deltaTime;
+                        if (regenData.NextRegenTimer <= 0)
+                        {
+                            regenData.NextRegenTimer += regenComponent.RegenInterval;
+
+                            // Send command to regen entity.
+                            var modifyHealthRequest = new HealthComponent.ModifyHealth.Request(
```

Comment:
should we be doing
```csharp
.Request(
	///
);
```

or

```csharp
.Request
(
	///
);
```

Reply:
Its formatted by the formatter this way.